### PR TITLE
Optimize feature transformer backpropagation stats.

### PR DESCRIPTION
**Pull request by Sopel97**
This PR optimizes gradient clipping and statistics gathering in the feature transformer backprogation method. It's part of an effort to optimize all sequential code in the trainer. I tested this with a short training run and didn't note any anomalies.

### CODE REVIEW

**Comment by nodchip** [comment]
The changes look good to me.  Thank you.
